From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: WayZer <himc.wicp@gmail.com>
Date: Wed, 6 Sep 2023 00:43:17 +0800
Subject: [PATCH] add `@server` to logic (#9018)

* add @master to logic

* remove `Vars.`

* Update GlobalVars.java

* Update GlobalVars.java

(cherry picked from commit c9beb3d158bee8d17fe4e057418710294f7eef65)
---
 core/src/mindustry/logic/GlobalVars.java | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/core/src/mindustry/logic/GlobalVars.java b/core/src/mindustry/logic/GlobalVars.java
index 3354277b130a3f62952734398882c980f883bab6..57b30bcd7bab28592d4970a5fb1fac7f3e41a58a 100644
--- a/core/src/mindustry/logic/GlobalVars.java
+++ b/core/src/mindustry/logic/GlobalVars.java
@@ -27,7 +27,7 @@ public class GlobalVars{
     public static final Rand rand = new Rand();
 
     //non-constants that depend on state
-    private static int varTime, varTick, varSecond, varMinute, varWave, varWaveTime;
+    private static int varTime, varTick, varSecond, varMinute, varWave, varWaveTime, varServer;
 
     private ObjectIntMap<String> namesToIds = new ObjectIntMap<>();
     private Seq<Var> vars = new Seq<>(Var.class);
@@ -56,6 +56,8 @@ public class GlobalVars{
         varWave = put("@waveNumber", 0);
         varWaveTime = put("@waveTime", 0);
 
+        varServer = put("@server", 0);
+
         //special enums
         put("@ctrlProcessor", ctrlProcessor);
         put("@ctrlPlayer", ctrlPlayer);
@@ -147,6 +149,9 @@ public class GlobalVars{
         //wave state
         vars.items[varWave].numval = state.wave;
         vars.items[varWaveTime].numval = state.wavetime / 60f;
+
+        //network
+        vars.items[varServer].numval = (net.server() || !net.active()) ? 1 : 0;
     }
 
     /** @return a piece of content based on its logic ID. This is not equivalent to content ID. */
